src/IJ_mv/IJMatrix_parcsr.c

```diff
@@ -2563,25 +2575,52 @@ hypre_IJMatrixAssembleParCSR(hypre_IJMatrix *matrix)
                ne = (my_thread_num+1)*size + rest;
             }
 
-            i_diag = 0;
-            i_offd = 0;
-            for (i=ns; i < ne; i++)
+            i_diag = i_offd = 0;
+            if (rownnz != NULL)
```

Comment:
It appears to me in many places we have this kind of `if ... else` where the main bodies of two branches are basically the same except that the if-branch uses `ii = rownnz[i]` and goes with `ii`, while the else-branch goes with `i`. I guess we may combine them in a single branch with something like `ii = rownnz ? rownnz[i] : i`, to reduce the amount of identical code.